test/test-sol.jl

```diff
@@ -4,5 +4,28 @@ solve_exp = Float64.(readdlm("out.csv", ',')[2:end, :])
 df = chamber(Silicic(), 1e9, 0.2, 0.04, 0.001, -3.3, 8e3)
 
 @testset "solve_ode" begin
-    @test all(abs.(solve_exp .- Matrix(df)) .< 0.00005)
+    for i in 1:11
+        println("col$i maximum diff: ", maximum!([1.0], abs.(solve_exp[:, i] .- Matrix(df)[:, i])))
+        @test all(abs.(solve_exp[:, i] .- Matrix(df)[:, i]) .< abs(solve_exp[end, i]*1e-9))
+    end
+end
+
+solve_exp1 = Float64.(readdlm("out_co2_1e-3.csv", ',')[2:end, :])
+solve_exp2 = Float64.(readdlm("out_co2_2e-3.csv", ',')[2:end, :])
+output_dirname = chamber(Mafic(), 1e9, 0.2, 0.01, [0.001, 0.002], -3.3, 8e3)
```

Comment:
```suggestion
output_dirname = chamber(Mafic(), 1e9, [0.2], [0.01], [0.001, 0.002], [-3.3], [8e3])
```

If you change the function signature to take vectors only, passing scalars for `log_volume_km3_vector`, `InitialConc_H2O_vector`, `InitialConc_CO2_vector`, `log_vfr_vector`, `depth_vector` will not work.

Reply:
Keeping the unions _might_ still work. I had two concerns.
1. I was afraid that the for loop wouldn't work when iterating over a pure float, but it does for some reason (?). 
```julia
julia> X = 1; Y = [2, 3];

julia> for x in X, y in Y
       @show (x, y)
       end
(x, y) = (1, 2)
(x, y) = (1, 3)

julia>
```
2. I wasn't sure how Julia would decide what method of the `chamber` function to call if all passed arguments were floats, but it seems to prioritize the method with the most concrete types.
```julia
julia> myfun(x::Float64, y::Float64) = "no unions"
myfun (generic function with 1 method)

julia> myfun(x::Union{Float64,Vector{Float64}}, y::Union{Float64,Vector{Float64}}) = "with unions"
myfun (generic function with 2 methods)

julia> myfun(1.0,[2.0])
"with unions"

julia> myfun(1.0, 2.0)
"no unions"

julia>
```

I guess keeping the unions is fine!